### PR TITLE
Move AID to traditional python threading

### PR DESCRIPTION
**Pull request by ivar-lazzaro**
As we become more dependent on multi-threading, Greenthread patch-and-pray libraries start behaving unpredictably, especially when running in containers (probably having something to do with the libc version). The most common problem we see is threads hanging indefinitely, bringing AID to an unrecoverable state unless restarted.
With this patch, we move AID to traditional python threading and introduce all the necessary concurrency management (eg. locking).

### CODE REVIEW

File: aim/agent/aid/universes/k8s/k8s_watcher.py [hunk 1]
```diff
@@ -179,40 +179,75 @@ def _observe_and_monitor_loop(self):
...
-            if tstate.get('http_resp') and tstate['http_resp'].closed:
-                LOG.info('HTTP response closed for thread %s', thd)
-                exc = K8SObserverStopped()
-                break
-            if thd.dead:
-                LOG.info('Thread %s is not alive', thd)
-                exc = K8SObserverStopped()
-                break
+            thd = tstate.get('thread')
+            LOG.info("Checking thread %s for %s" % (thd, t))
```
Comment (amittbose) [inline aim/agent/aid/universes/k8s/k8s_watcher.py]: this line will make logs very chatty

File: aim/agent/aid/universes/k8s/k8s_watcher.py [hunk 2]
```diff
@@ -179,40 +179,75 @@ def _observe_and_monitor_loop(self):
...
+                if tstate.get('watch_exception'):
+                    exc = tstate.get('watch_exception')
+                    LOG.info('Thread %s raised exception %s', thd, exc)
+                    break
+                if tstate.get('http_resp') and tstate['http_resp'].closed:
+                    LOG.info('HTTP response closed for thread %s', thd)
+                    exc = K8SObserverStopped()
+                    break
+                else:
+                    LOG("https resp for %s: %s" % (t, tstate.get('http_resp')))
```
Comment (amittbose) [inline aim/agent/aid/universes/k8s/k8s_watcher.py]: LOG without a level? Should this even be logged?

File: aim/tests/unit/test_aim_manager.py [hunk 3]
```diff
@@ -1490,9 +1490,9 @@ class TestVmmInjectedHostMixin(object):
                                 'domain_name': 'kubernetes',
                                 'controller_name': 'kube-cluster',
                                 'host_name': 'host1.local.lab',
-                                'os': 'Ubuntu',
+                                'os': 'Ubunt',
```
Comment (amittbose) [inline aim/tests/unit/test_aim_manager.py]: why is this required?

File: aim/agent/aid/universes/k8s/k8s_watcher.py (hunk 1 above)
Comment (ivar-lazzaro) [inline aim/agent/aid/universes/k8s/k8s_watcher.py]: oh this was something I did for debug and somehow ended up in this branch, will clean up

File: aim/agent/aid/universes/k8s/k8s_watcher.py (hunk 2 above)
Comment (ivar-lazzaro) [inline aim/agent/aid/universes/k8s/k8s_watcher.py]: same as above

File: aim/tests/unit/test_aim_manager.py (hunk 3 above)
Comment (ivar-lazzaro) [inline aim/tests/unit/test_aim_manager.py]: replaced u' with ' I guess, will rollback

File: aim/agent/aid/universes/k8s/k8s_watcher.py [hunk 4]
```diff
@@ -229,10 +229,15 @@ def wrapped_list(f, k8s_type, *args, **kwargs):
...
         exc = None
+        RESTART_TIME = 28 * 60
+        if not getattr(self, '_check_time', None):
+            self._check_time = time.time()
         for t in self._observe_thread_state:
             tstate = self._observe_thread_state[t]
             thd = tstate.get('thread')
-            if thd:
+            if time.time() - self._check_time > RESTART_TIME:
+                exc = K8SObserverStopped()
```
Comment (amittbose) [inline aim/agent/aid/universes/k8s/k8s_watcher.py]: add a break here?
